common.blocks/i-bem/__dom/i-bem__dom.js

```diff
@@ -736,13 +736,23 @@ DOM = BEM.decl('i-bem__dom',/** @lends BEMDOM.prototype */{
             modName = undef;
         }
 
+        names = names.split(' ');
+
         var _self = this.__self,
-            selector = '.' +
-                names.split(' ').map(function(name) {
-                    return _self.buildClass(name, modName, modVal);
-                }).join(',.'),
+            keys = names.map(function(name) {
+                return _self.buildClass(name, modName, modVal);
+            }),
+            isSingleName = keys.length === 1,
+            selector = '.' + (isSingleName? keys[0] : keys.join(',.')),
             res = findDomElem(ctx, selector);
 
+        // caching results if possible
+        (ctx === this.domElem) && keys.forEach(function(key, i) {
+                var elem = this._elemCache[key] = isSingleName? res : res.filter('.' + key);
+                delete elem.prevObject; // drop link to original set after filtering
```

Comment:
Fixed, added test. Check please.

There is `.is` function but I'm not sure about we should use it. It's pretty slow in compare with raw equal (but still much more readable): http://jsperf.com/jquery-objects-comparison